### PR TITLE
Add ltss repos in ltss guest VM

### PR DESCRIPTION
**Pull request by tonyyuan1**
Logic:
1. Check if the guest OS is ltss by checking ltss scc code is present. (Each sle version has its own ltss code)
2.  Register ltss scc repo if guest OS is ltss.

This requires a hash to store ltss code. The hash should not be made public in github.
Therefore, it can be placed nowhere but job settings.  A variable below can be easily converted to hash

LTSS_PRODUCTS_CODES="15.1,INTERNAL-USE-xxxxxx,15,INTERNAL-USE-ONLY-yyyyyyy,12.4,INTERNAL-USE-ONLY-cccccc,12.3,INTERNAL-USE-ONLY-dddddd".

See the job settings of VR below for "LTSS_PRODUCTS_CODES" example.

- Related ticket: https://progress.opensuse.org/issues/106793
- Needles: no
- Verification run: 
- sles12sp4 kvm: http://openqa.qam.suse.cz/tests/37180#step/register_guests/8
- sles15 xen: http://openqa.qam.suse.cz/tests/37201#step/register_guests/20
- sles15sp1 kvm: http://openqa.qam.suse.cz/tests/37263#step/register_guests/8
- sle15SP2 kvm: http://openqa.qam.suse.cz/tests/37495#step/register_guests/19      VR after change suggested by reviewers

### CODE REVIEW

File: tests/virtualization/universal/register_guests.pm [hunk 1]
```diff
@@ -16,13 +16,20 @@ use version_utils;
...
+    select_console 'root-console';
+    my %ltss_products = @{get_var_array("LTSS_PRODUCTS")};
 
     foreach my $guest (keys %virt_autotest::common::guests) {
         record_info "$guest", "Registrating $guest against SMT";
         my ($sles_running_version, $sles_running_sp) = get_os_release("ssh root\@$guest");
+        my $version_id = $sles_running_sp != 0 ? "$sles_running_version.$sles_running_sp" : $sles_running_version;
         if ($sles_running_version >= 12) {
             assert_script_run("ssh root\@$guest SUSEConnect -r " . get_var('SCC_REGCODE') . " -e " . get_var("SCC_EMAIL"));
+            sleep 10;
```
Comment (Julie-CAO) [inline tests/virtualization/universal/register_guests.pm:29]: Is sleep necessary? if it is required, a comment is welcome as it is not recommended in openqa coding rule.

File: tests/virtualization/universal/register_guests.pm [hunk 2]
```diff
@@ -16,13 +16,20 @@ use version_utils;
 sub run_test {
     my ($self) = @_;
     # Use serial terminal, unless defined otherwise. The unless will go away once we are certain this is stable
-    $self->select_serial_terminal unless get_var('_VIRT_SERIAL_TERMINAL', 1) == 0;
+    # $self->select_serial_terminal unless get_var('_VIRT_SERIAL_TERMINAL', 1) == 0;
+    select_console 'root-console';
+    my %ltss_products = @{get_var_array("LTSS_PRODUCTS")};
```
Comment (Julie-CAO) [inline tests/virtualization/universal/register_guests.pm:21]: the setting name, LTSS_PRODUCTS, is a little confusing. It looks like the product list instead of scc codes.

**Comment by Julie-CAO** [comment]
I made comments about minnor issues, others LGTM.

File: tests/virtualization/universal/register_guests.pm [hunk 3]
```diff
@@ -16,13 +16,20 @@ use version_utils;
...
     foreach my $guest (keys %virt_autotest::common::guests) {
         record_info "$guest", "Registrating $guest against SMT";
         my ($sles_running_version, $sles_running_sp) = get_os_release("ssh root\@$guest");
+        my $version_id = $sles_running_sp != 0 ? "$sles_running_version.$sles_running_sp" : $sles_running_version;
         if ($sles_running_version >= 12) {
             assert_script_run("ssh root\@$guest SUSEConnect -r " . get_var('SCC_REGCODE') . " -e " . get_var("SCC_EMAIL"));
+            sleep 10;
+            assert_script_run("ssh root\@$guest SUSEConnect -p SLES-LTSS/$version_id/x86_64 -r " . $ltss_products{"$version_id"}) if ($ltss_products{"$version_id"});
+            sleep 10;
+            record_info($ltss_products{"$version_id"});
```
Comment (Julie-CAO) [inline tests/virtualization/universal/register_guests.pm:32]: Do we need the scc codes of LTSS products is shown so noticeable? Of course, this may be not a problem just preference.

File: tests/virtualization/universal/register_guests.pm (hunk 2 above)
Comment (alice-suse) [reply tests/virtualization/universal/register_guests.pm:21]: @tonyyuan1  Agree, better to rename the variable. Besides, where do you put the setting? In testsuite qam-kvm-install and qam-xen-install's settings, rather than job group or medium settings, right?

File: tests/virtualization/universal/register_guests.pm (hunk 3 above)
Comment (alice-suse) [reply tests/virtualization/universal/register_guests.pm:32]: @tonyyuan1  Agree, we'd better not explicitly show it in record_info. And even better, you can rename LTSS_PRODUCTS to LTSS_PRODUCTS_REGCODE_SECRET, so that in jobs' vars.json and settings webpage, it will not be displayed.

File: tests/virtualization/universal/register_guests.pm [hunk 4]
```diff
@@ -16,13 +16,20 @@ use version_utils;
...
+    # $self->select_serial_terminal unless get_var('_VIRT_SERIAL_TERMINAL', 1) == 0;
+    select_console 'root-console';
+    my %ltss_products = @{get_var_array("LTSS_PRODUCTS")};
 
     foreach my $guest (keys %virt_autotest::common::guests) {
         record_info "$guest", "Registrating $guest against SMT";
         my ($sles_running_version, $sles_running_sp) = get_os_release("ssh root\@$guest");
+        my $version_id = $sles_running_sp != 0 ? "$sles_running_version.$sles_running_sp" : $sles_running_version;
         if ($sles_running_version >= 12) {
             assert_script_run("ssh root\@$guest SUSEConnect -r " . get_var('SCC_REGCODE') . " -e " . get_var("SCC_EMAIL"));
```
Comment (alice-suse) [inline tests/virtualization/universal/register_guests.pm:28]: @tonyyuan1  This comment is not related to your PR. But do you find this line is registering wrongly too? The SCC_REGCODE is the correct regcode for the product that the host is installed to, rather than all vms which mostly are different from host version. HDYT?

Comment (Julie-CAO) [reply tests/virtualization/universal/register_guests.pm:28]: I found the values of SCC_REGCODE for different product were identical in OSD, but varied in arch'es. So the SCC_REGCODE from host may work for most guests except LTSS ones. It is just my observation, not sure if it is correct.

File: tests/virtualization/universal/register_guests.pm (hunk 1 above)
Comment (tonyyuan1) [reply tests/virtualization/universal/register_guests.pm:29]: Sometime "SUSEConnect -r" takes a bit longer time to complete. I once saw it broke the next "SUSEConnect -r ltss" . I don't why that happened. I am not sure if it necessary for the moment,   just feel that it's safe to wait a few seconds to perform next register. 
I will add a comment there.

File: tests/virtualization/universal/register_guests.pm (hunk 2 above)
Comment (tonyyuan1) [reply tests/virtualization/universal/register_guests.pm:21]: It includes info about both product and scc code .  The key represents product and its value is ltss scc code. A key is used to check if a guest os is ltss product. If yes, register with the corresponding code. 

Is LTSS_PRODUCTS reasonable? or any better one?

Yes,  In testsuite qam-kvm-install and qam-xen-install's settings. It is only used by the two.

File: tests/virtualization/universal/register_guests.pm (hunk 4 above)
Comment (tonyyuan1) [reply tests/virtualization/universal/register_guests.pm:28]: Yes, Julie answered for me :)

Comment (alice-suse) [reply tests/virtualization/universal/register_guests.pm:28]: Okay, that's fine

File: tests/virtualization/universal/register_guests.pm (hunk 3 above)
Comment (tonyyuan1) [reply tests/virtualization/universal/register_guests.pm:32]: They are key test data in this test.  It's helpful to find out some hiding errors which don't even fail/soffail test. 
It might also help reviewers/QA to understand this test quickly to save their time. 
Yes, it's not good to scc code. I will remove them.

Isn't LTSS_PRODUCTS_REGCODE_SECRET too lengthy?

Comment (alice-suse) [reply tests/virtualization/universal/register_guests.pm:32]: Yes, a little bit, but SECRET is necessary to hide it, how about LTSS_REGCODE_SECRET? Up to you to decide better ones.

Comment (tonyyuan1) [reply tests/virtualization/universal/register_guests.pm:32]: LTSS_REGCODES_SECRET is fine

**Comment by alice-suse** [comment]
LGTM. Let's also invite @tbaev and @martinsmarcelo for review.

**Comment by alice-suse** [comment]
@tbaev Would you like to have a review?

**Review by tbaev — APPROVED** [review]

**Comment by tbaev** [comment]
Looks good to me. :)
